src/rootPages/Designer/ui_work_interface_list_copyPage.js

```diff
@@ -0,0 +1,453 @@
+                     name: "name",
+                     required: true,
+                     placeholder: `${oldName} - ${L("copy")}`,
+                     labelWidth: 110,
+                     on: {
+                        onAfterRender() {
+                           AB.ClassUI.CYPRESS_REF(
+                              this,
+                              "ui_work_interface_list_newPage_blank_name"
+                           );
+                        },
+                     },
+                  },
+                  {
+                     margin: 5,
+                     cols: [
+                        { fillspace: true },
+                        {
+                           view: "button",
+                           id: this.ids.buttonCancel,
+                           value: L("Cancel"),
+                           css: "ab-cancel-button",
+                           autowidth: true,
+                           click: () => {
+                              this.cancel();
+                           },
+                           on: {
+                              onAfterRender() {
+                                 AB.ClassUI.CYPRESS_REF(this);
+                              },
+                           },
+                        },
+                        {
+                           view: "button",
+                           id: this.ids.buttonSave,
+                           css: "webix_primary",
+                           value: L("Paste Page"),
+                           autowidth: true,
+                           type: "form",
+                           click: () => {
+                              return this.save();
+                           },
+                           on: {
+                              onAfterRender() {
+                                 AB.ClassUI.CYPRESS_REF(this);
+                              },
+                           },
+                        },
+                     ],
+                  },
+               ],
+            },
+            on: {
+               onBeforeShow: () => {
+                  // var id = $$(this.ids.tab).getValue();
+                  // this.switchTab(id);
+               },
+            },
+         };
+      }
+
+      async init(AB, data) {
+         this.AB = AB;
+         this.data = data;
+         console.log(data);
+
+         webix.ui(this.ui(data.name));
+         webix.extend($$(this.ids.component), webix.ProgressBar);
+
+         this.$form = $$(this.ids.form);
+
+         this.$component = $$(this.ids.component);
+
+         // if there was an error saving the values from our form.
+         this.on("save.error", (err) => {
+            this.onError(err);
+         });
+
+         // if the values we provided were successfully saved.
+         this.on("save.successful", () => {
+            this.onSuccess();
+         });
+
+         // init() routines are always considered async so:
+         return Promise.resolve();
```

Comment:
technically, since you declared the `fn()`  as `async`, we don't need to return a promise here.

Reply:
removed